### ISSUE TITLE
getRows returns empty array

### ISSUE THREAD

**Issue by joshua-mbg** [theoephraim/node-google-spreadsheet#200]
`bash-4.4$ npm install google-spreadsheet
npm WARN deprecated node-uuid@1.4.8: Use uuid module instead
npm WARN sandbox@1.0.0 No description
npm WARN sandbox@1.0.0 No repository field.

+ google-spreadsheet@2.0.4
removed 5 packages and updated 2 packages in 9.986s
bash-4.4$ node -v && npm -v
v9.4.0
5.6.0
`

`
docinfo.worksheets.forEach((worksheet, index) => {
  doc.getRows(index+1,{},(err, rowData) => { console.log(rowData); });
});
// []

docinfo.worksheets.forEach((worksheet, index) => {
  worksheet.getRows({},(err, rowData) => { console.log(rowData); });
});
// []
`

✓ SymbolAuthentication works. 
✓ Does access document correctly. 
✓ Does return docinfo
✓ Does return worksheets
✘ Does not get rows of worksheets with data in them

**Comment by lynxaegon** [comment]
+1

**Comment by tomasbruckner** [comment]
same

**Comment by tomasbruckner** [comment]
@theoephraim yeah, it was the same problem. I am not sure how to make it more developer friendly to be honest. The thing that confused me the most was that rowCount returned correct number and there was a filter used in my google sheet (that I didn't see) so the first n blank rows were hidden. So at the first sight I thought that there were no blank rows.

I know it is row API limitation, but it is just somewhat confusing that spreadsheet will tell me it has 100 rows and I call `getRows` with `rowCount: 100` and it returns empty array.
